### PR TITLE
Stop the user from losing changes to a schedule

### PR DESCRIPTION
**Pull request by Guntrisoft**
- If the user makes any changes and doesn't hit the save
button then they will be alerted that they have changes
that will be lost

### CODE REVIEW

File: app/assets/javascripts/calendars/guider-slot-picker.es6 [hunk 1]
```diff
@@ -20,22 +20,44 @@
...
+    }
+
     select(start) {
       const event = {
         start: start,
         end: moment(start).add(this.config.slotDurationMinutes, 'minutes')
       };
 
       if (!this.isOverlapping(event)) {
+        this.clearUnloadEvent();
```
Comment (AndrewVos) [inline app/assets/javascripts/calendars/guider-slot-picker.es6]: Why is this here?

File: app/assets/javascripts/calendars/guider-slot-picker.es6 [hunk 2]
```diff
@@ -44,9 +66,14 @@
     }
 
     eventRender(event, element) {
+      if ($.inArray(event.start.unix(), this.initialEventTimestamps) === -1) {
+        this.setUnloadEvent();
+      }
+
```
Comment (AndrewVos) [inline app/assets/javascripts/calendars/guider-slot-picker.es6]: Oh wow, this is beautiful!

File: app/assets/javascripts/calendars/guider-slot-picker.es6 (hunk 1 above)
Comment (Guntrisoft) [inline app/assets/javascripts/calendars/guider-slot-picker.es6]: We need to reset the unload event, any time something happens to the calendar

Comment (AndrewVos) [inline app/assets/javascripts/calendars/guider-slot-picker.es6]: Oh right I see
